scripts/generate_reference.py

```diff
@@ -50,10 +54,13 @@ def solver_config(study_name):
 
     # Do we need named MPS problems ?
```

Comment:
This part of the script (main part) does not properly divide the work the script does into steps.
To me, what this script does is :
- processes the exe options
- run the executable
- move the results to a **reference** directory

Here we have too much details for processing the script arguments, a no detail at all about moving the results into a **reference** directory. We should encapsulate the details.

For instance, we could have something like :
```py
for study_path in study_collection:
    run_command_factory RunCommandFactory(script_args, study_path)
    command_to_run = run_command_factory.get_cmd()
    if not run_binary(command_to_run) : 
        return False
    return True
    
    move_results_to_reference(study_path)
```
Not that first and last stage would surely require polymorphism, as they differ depending on we run **solver** exe, **ts-generator** exe or want to save (to reference dir) mps files, TS files or classic results.